### ISSUE TITLE
Sort out -manual filename convention vs meta element "interact" flag

### ISSUE THREAD

**Issue by annevk** [w3c/web-platform-tests#5381]
Both doesn't seem like a good long term solution. The filename convention seems a little clearer.

Inspired by #5370.

cc @zcorpan @mrego @frivoal @fantasai @jgraham

**Comment by gsnedders** [comment]
The meta element flag should cause a lint failure anywhere outside of `css` (or rather, a flag that causes them to become manual *without* the filename also ending in `-manual` is a lint failure).

The reason why it's in `css` is to avoid mass-renaming files in there and breaking all existing results in the CSS test harness. Potentially, we could alter the build system such that tests with `-manual` or `-visual` got that stripped and the flag added as part of the build system, but that requires someone to do that.

**Comment by frivoal** [comment]
I agree that one convention would be much better than two. That said, no strong feelings from me in favor of naming vs flags. As long as it is clearly documented, and supported by the infrastructure (including the CSS test harness), I'm happy with either.

**Comment by fred-wang** [comment]
Are all the manual tests supposed to use either "interact" flag or -manual filename convention?

I had to add -manual suffix for CSS animations tests ( https://github.com/web-platform-tests/wpt/pull/11215 ) ; these tests use the "animation" flag but not the "interact" one...

cc @youennf

**Comment by frivoal** [comment]
as of today, there isn't a single flag for non-automatic tests. There are several ones, depending on why it cannot be automated:
* interact (the tester needs to do something)
* animated (the test is animated in final state)
* font (Requires a specific font to be installed at the OS level)
* userstyle (requires the tester to install a particular user stylesheet)

This situation is not good, but that's where we are for now.

We need to do two things:
* decide which convention we want to use going forward (-manual suffix seems nice to me)
* decide whether to convert existing tests, or to deprecate but continue to support old conventions. (convert seems ok with me, but we need to be really sure, as renaming thousands of files isn't nice on systems that have metadata linked to filenames).

**Comment by annevk** [comment]
Some thoughts:

1. It seems that outside `/css` we already decided upon `-manual`.
2. Renaming lots of files has been done quite a bit (at least outside `/css`). This mostly went fine (updating links can be a pain).
3. If you want to postpone renaming you could safelist all current usage and forbid further flag usage with the lint script.

**Comment by gsnedders** [comment]
> If you want to postpone renaming you could safelist all current usage and forbid further flag usage with the lint script.

FWIW, we currently have `css/*` whitelisted for the `CONTENT-MANUAL` (and the related `CONTENT-VISUAL`) lint.

> as of today, there isn't a single flag for non-automatic tests. There are several ones, depending on why it cannot be automated:

The canonical list for what makes them manual is `content_is_css_manual` in sourcefile.py: https://github.com/web-platform-tests/wpt/blob/master/tools/manifest/sourcefile.py#L537

Currently, this is:

```{"animated", "font", "history", "interact", "paged", "speech", "userstyle"}```

**Comment by gsnedders** [comment]
As for each of those:

* **animated**: these I don't think anyone is going to have automation testing, especially in a cross-browser way, hence I don't think it's worthwhile distinguishing these
* **font**: these can be automated with a manual step (installing the font), but it's probably not worthwhile adding links to the fonts in some machine readable form so they can be run automatically; the vast majority can be moved over to using web fonts (see also #9374)
* **history**: I have some memory of this being here because of there being tests that required "open this other page then reload" but I can't find any that don't also have interact, so maybe it should just go from the set
* **interact**: obviously
* **paged**: nothing per-se keeping it here except for #7760; if we want to do this, it might be worth adding some way to mark these up even if wptrunner doesn't run them
* **speech**: I don't think there's ever been interest from implementors in running these, and it's unclear how one would test them?
* **userstyle**: like font these can be automated with a manual step (installing the user stylesheet), but whether it's worthwhile working on depends on whether we expect implementations to keep on shipping support for user stylesheets (several having already dropped support)

**Comment by zcorpan** [comment]
> The reason why it's in css is to avoid mass-renaming files in there and breaking all existing results in the CSS test harness.

@plinss has fixed this, so this should unblock the renaming.

I suggest that we keep the metadata when doing the renaming, however when authoring new tests it would be good if it was sufficient to use only the file name convention. I'm not sure if the CSS test harness or something needs further changes to support that.

**Comment by zcorpan** [comment]
Moreover, we should update the linter/whitelist to complain about any new test that is manual according to `flags` but doesn't have `-manual`.

**Comment by plinss** [comment]
Yes, please keep the rest of the existing metadata for now. I can have the test harness extract the additional flags it needs from the filenames at some point, but it's not doing that yet.

**Comment by gsnedders** [comment]
@zcorpan FWIW, still in favour of splitting out paged reftests into a new test type before we do the rename, rather than ending up renaming twice. (#7760)

Adding the existing flags-based manual tests to the whitelist might regress performance of the lint (given it's a huge number of extra files to whitelist), unless we change how we do the whitelisting.

@plinss Does it just use `testinfo.data` for it? I presume the logic is something like "if there's a interact flag, it's manual; if there's a script flag, it's a script test; if there's a reference, it's a reftest; otherwise it's a visual test"? Then one option is just make the build system do that.

It also strikes me having the harness change the ids we need to have the lint deal with `-manual` and -visual` when it comes to avoiding id clashes (CSS-COLLIDING-*).

**Comment by plinss** [comment]
No changes were made to the testinfo.data for this so far, the only change was in the harness test import to filter off the -manual and -visual suffixes from the test names.

The harness generally presumes all tests are manual regardless of flags set in the testinfo.data, the only one it really cares about IIRC is 'reference' but it auto-sets that one based on the presence of references (or the build does). It does detect the testharness script automatically if the 'auto-submit' switch is turned on during a run. The other flags are displayed to the user and can be used for filtering test runs, but aren't otherwise significant except for the 'may' and 'should' which are used for computing if CR exit criteria are met.

And yes, since the suffixes are stripped from the test ids then there's a potential for conflicts.

**Comment by zcorpan** [comment]
@gsnedders I won't have time to follow through with the renaming, and I'll be away the next 7 weeks. Can you take this?

**Comment by Ms2ger** [comment]
Can whoever changed the CSS thing also change the lint?

**Comment by stephenmcgruer** [comment]
Is the current advice here still that renaming is fine, but that meta flags should be kept? Context, I was just added as a reviewer on https://github.com/web-platform-tests/wpt/pull/18978 (well, on the upstream Chromium CL)

**Comment by gsnedders** [comment]
@stephenmcgruer yes

**Comment by gsnedders** [comment]
For those following along here, I posted https://github.com/web-platform-tests/rfcs/pull/41 so we can come up with some plan when it comes to what we're doing about paged media tests, so we can then sort this out.

**Comment by gsnedders** [comment]
> [I'm] in favour of splitting out paged reftests into a new test type before we do the rename, rather than ending up renaming twice.

This has now come to pass, so it'd be nice to try and do this sometime soon. I guess we need an RFC at this point (it's a sufficiently large change), but that doesn't seem too problematic.

If anyone wants to take stab at moving paged tests to reftests for those that aren't already, there's a list at https://gist.github.com/gsnedders/cf0ac1e1b97f6955ff530a683ee6481d (note this naïvely just looks for flags=paged and doesn't check which are/aren't tests, and it looks like a few of the files in there are references already).

**Comment by DanielRyanSmith** [comment]
Hello everyone!

I've been taking a look at issue #23253 and it has taken me down a lot of paths for handling some of the old flags used that are now deprecated in favor of the `-manual` file suffix. I created a [draft PR](https://github.com/web-platform-tests/wpt/pull/32134) that adds this suffix to any file in the `css` directory does not already have the suffix and contains any of the flags `interact`, `animated`, `history`, and `userstyle`. I was hoping this could be used to eventually remove the old flags and then create an additional lint check to see if a user is attempting to add the deprecated flags and advise them to use the correct file name instead.

Is this mass-filename change a valid and useful approach? Are there any steps or details that I'm missing? Additionally, am I understanding correctly that `speech` will be handled differently rather than adding the `-manual` suffix to these files? Thanks for the help!

**Comment by frivoal** [comment]
@plinss Is there anything that can be done to avoid test results on https://test.csswg.org/harness/ being lost as a result of the rename?

**Comment by gsnedders** [comment]
> [@plinss](https://github.com/plinss) Is there anything that can be done to avoid test results on https://test.csswg.org/harness/ being lost as a result of the rename?

This was [done in 2018](https://hg.csswg.org/dev/harness/rev/b6acc85fa820), [as he commented above](https://github.com/web-platform-tests/wpt/issues/5381#issuecomment-397849476).
